simexpal/base.py

```diff
@@ -1128,26 +1159,71 @@ def output_file_path(self, ext):
 		return os.path.join(self.experiment.output_subdir,
 				get_output_file_name(ext, self.instance.shortname, self.repetition))
 
-	def get_status(self):
-		if os.access(self.output_file_path('status'), os.F_OK):
-			with open(self.output_file_path('status'), "r") as f:
+	def _update_status_cache_dict(self):
+
+		# Determine status cache entry.
+		status, last_mod = Status.NOT_SUBMITTED, None
+		try:
+			with open(self.output_file_path('status'), 'r') as f:
 				status_dict = yaml.load(f, Loader=YmlLoader)
+				last_mod = os.fstat(f.fileno()).st_mtime
 
 			if status_dict['timeout']:
-				return Status.TIMEOUT
+				status = Status.TIMEOUT
 			elif status_dict['signal']:
-				return Status.KILLED
+				status = Status.KILLED
 			elif status_dict['status'] > 0:
-				return Status.FAILED
-			return Status.FINISHED
-		elif os.access(self.output_file_path('out'), os.F_OK):
-			return Status.STARTED
-		elif os.access(self.aux_file_path('run'), os.F_OK):
-			return Status.SUBMITTED
-		elif os.access(self.aux_file_path('lock'), os.F_OK):
-			return Status.IN_SUBMISSION
-
-		return Status.NOT_SUBMITTED
+				status = Status.FAILED
+			else:
+				status = Status.FINISHED
+		except FileNotFoundError:
+			if os.access(self.output_file_path('out'), os.F_OK):
+				status, last_mod = Status.STARTED, os.stat(self.output_file_path('out')).st_mtime
+			elif os.access(self.aux_file_path('run'), os.F_OK):
+				status, last_mod = Status.SUBMITTED, os.stat(self.aux_file_path('run')).st_mtime
+			elif os.access(self.aux_file_path('lock'), os.F_OK):
+				status, last_mod = Status.IN_SUBMISSION, os.stat(self.aux_file_path('lock')).st_mtime
+
+		internal_name = self.internal_name
+		if internal_name not in self._cfg.status_cache_dict:
+			self._cfg.status_cache_dict[internal_name] = {
+				self.instance.shortname: {'status': status, 'last_mod': last_mod}}
```

Comment:
Is `shortname` enough to uniquely identify an instance?

Reply:
It should be enough. When constructing the instance dictionary ``Config._insts`` we check for instances that resolve to the same ``shortname`` (and raise an error in case it does)

Reply:
Alright.